doc/api/assert.md

```diff
@@ -13,6 +13,71 @@ A `strict` and a `legacy` mode exist, while it is recommended to only use
 For more information about the used equality comparisons see
 [MDN's guide on equality comparisons and sameness][mdn-equality-guide].
 
+## Class: AssertionError
+
+A subclass of `Error` that indicates the failure of an assertion.
+All errors thrown by the assert module will be instance of the `AssertionError`
+class.
+
+### new AssertionError(options)
+<!-- YAML
+added: v0.1.21
+-->
+* `options` {Object}
+  * `message` {string} If provided, the error message is going to be set to this
+    value.
+  * `actual` {any} The `actual` property on the error instance is going to
+    contain this value. Internally used for the `actual` error input in case
+    e.g., [`assert.strictEqual()`] is used.
+  * `expected` {any} The `expected` property on the error instance is going to
+    contain this value. Internally used for the `expected` error input in case
+    e.g., [`assert.strictEqual()`] is used.
+  * `operator` {string} The `operator` property on the error instance is going
+    to contain this value. Internally used to indicate what operation was used
+    for comparison (or what assertion function triggered the error).
+  * `stackStartFn` {Function} If provided, the generated stack trace is going to
+    remove all frames up to the provided function.
+
+A subclass of `Error` that indicates the failure of an assertion.
+
+All instances contain the built-in Error properties (i.e., `message` and `name`)
+plus the following:
+
+* `actual` {any} Set to the actual value in case e.g.,
+  [`assert.strictEqual()`] is used.
+* `expected` {any} Set to the expected value in case e.g.,
+  [`assert.strictEqual()`] is used.
+* `generatedMessage` {boolean} Indicates if the message was auto generated
+  (`true`) or not.
+* `code` {string} This is always set to the string `ERR_ASSERTION` to indicate
+  that the error is actually a assertion error.
+* `operator` {string} Set to the passed in operator value.
+
+```js
+const assert = require('assert');
+
+// Generate a AssertionError to compare the error message later:.
+const { message } = new assert.AssertionError({
+  actual: 1,
+  expected: 2,
+  operator: '=='
+});
+
+// Verify error output:
+try {
+  assert.equal(1, 2);
+} catch (err) {
+  assert(err instanceof assert.AssertionError);
+  assert.equal(err.message, message);
+  assert.equal(err.name, 'AssertionError [ERR_ASSERTION]');
+  assert.equal(err.actual, 1);
+  assert.equal(err.expected, 2);
+  assert.equal(err.code, 'ERR_ASSERTION');
+  assert.equal(err.operator, '==');
+  assert.equal(err.generatedMessage, true);
```

Comment:
Not sure if we should popularize legacy loose equality mode here.
Mabe this way?
```js
const assert = require('assert');

// Generate an AssertionError to compare the error message later:
const { message } = new assert.AssertionError({
  actual: 1,
  expected: 2,
  operator: 'strictEqual'
});

// Verify error output:
try {
  assert.strictEqual(1, 2);
} catch (err) {
  assert(err instanceof assert.AssertionError);
  assert.strictEqual(err.message, message);
  assert.strictEqual(err.name, 'AssertionError [ERR_ASSERTION]');
  assert.strictEqual(err.actual, 1);
  assert.strictEqual(err.expected, 2);
  assert.strictEqual(err.code, 'ERR_ASSERTION');
  assert.strictEqual(err.operator, 'strictEqual');
  assert.strictEqual(err.generatedMessage, true);
}
```
Or maybe `1` and `'1'`?